### PR TITLE
RTECO-1362: Add comprehensive npm fail-on-uncollected-deps integration tests

### PR DESCRIPTION
**Pull request by udaykb2**
## Summary
Integration coverage for `--fail-on-uncollected-deps`. Test-only plus dependency bumps; no extra production code.
- **Healthy install** (`TestNpmFailOnUncollectedDeps`): no flag, each type, combos, semantic exclusion. Proves no false fail on `npmproject`.
- **Validation** (`TestNpmFailOnUncollectedDepsNegative`): unknown values, case, trailing/leading commas, `all` mixed with a type.
- **Requires build-info** (`TestNpmFailOnUncollectedDepsRequiresBuildInfo`): flag without `--build-name`/`--build-number` is rejected.
- **Regular / optional miss** (`TestNpmFailOnUncollectedDepsErrorFormat`): isolated cache → wipe tarballs → reinstall. Cache wipe is regular/optional only (checksum path).
- **Bundle miss** (`TestNpmFailOnUncollectedDepsBundle`): pack a local child that bundles `xml`, depend on the generated `.tgz` (not checked in). `npm ls` reports `_inBundle` + empty integrity. Verified locally against bukgradlefix.
- **Legacy** (`TestNpmMissingDepsLegacyBehavior`): same cache wipe without the flag still succeeds.
Peer is not e2e here: `PeerMissing` is npm 6; CLI npm CI is Node 16 / npm 8.
Pins: `build-info-go` @ `ea18305`, `jfrog-cli-artifactory` @ `3d48941`.

## Test Summary

| Category | Count | Status |
|----------|-------|--------|
| Positive | 16 | ✅ Pass |
| Negative | 7 | ✅ Pass |
| Error Format | 7 | ✅ Pass |
| Legacy | 1 | ✅ Pass |
| **Total** | **31** | **✅ All Pass** |

## Running Tests

```bash
# All tests
go test -v -run "TestNpmFailOnMissingDeps|TestNpmMissingDeps" \
  -test.npm=true -timeout 10m

# Individual categories
go test -v -run TestNpmFailOnMissingDepsPositive -test.npm=true -timeout 10m
go test -v -run TestNpmFailOnMissingDepsNegative -test.npm=true -timeout 10m
go test -v -run TestNpmFailOnMissingDepsErrorFormat -test.npm=true -timeout 10m
go test -v -run TestNpmMissingDepsLegacyBehavior -test.npm=true -timeout 10m
```

## Impact

- ✅ 31 comprehensive tests added
- ✅ Error scenarios use actual cache corruption (not mocked)
- ✅ Backward compatibility verified
- ✅ No production code changes
- ✅ Very low risk (test-only additions)

**Ready to merge** after previous PRs ✅

- [ ] All [tests](https://github.com/jfrog/jfrog-cli/blob/master/CONTRIBUTING.md#tests) have passed. If this feature is not already covered by the tests, new tests have been added.
- [ ] The pull request is targeting the `master` branch.
- [ ] The code has been validated to compile successfully by running `go vet ./...`.
- [ ] The code has been formatted properly using `go fmt ./...`.

---

### CODE REVIEW

File: testdata/npm/npmfailonmissingdeps/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "peerDependencies": {
+    "react": "^16.8.0 || ^17.0.0 || ^18.0.0"
+  },
+  "optionalDependencies": {
+    "sharp": "^0.32.1"
+  },
+  "bundleDependencies": [
+    "lodash"
+  ]
+}
```
Comment (fluxxBot) [inline testdata/npm/npmfailonmissingdeps/package.json]: can we comment which dependencies are not correct?
if possible can we use the existing test package.json's?

**Review by fluxxBot — APPROVED** [review]

File: npm_test.go [hunk 2]
```diff
@@ -1686,3 +1686,634 @@ func TestNpmPublishWithLocalGitVcsProps(t *testing.T) {
...
+// - 8 permutations/combinations of 2+ flags
+// - 3 semantic edge cases verifying exclusion logic
+// Total: 15 subtests covering all realistic success scenarios
+//
+// FAILURE PATHS (tested in build-info-go unit tests):
+// - TestHandleFailOnMissingDeps verifies all flag/depType combinations trigger correct failures
+// - All 4 dependency types: peer, optional, regular, bundle
+// - All flag combinations tested with proper mocking
+// - See: build-info-go/build/utils/npm_test.go line 816+
+func TestNpmFailOnMissingDeps(t *testing.T) {
```
Comment (bhanurp) [inline npm_test.go]: which scenario is actually causing the missing dependency in npm cache?

File: testdata/npm/npmfailonmissingdeps/package.json (hunk 1 above)
Comment (udaykb2) [reply testdata/npm/npmfailonmissingdeps/package.json]: all dependencies are correct.

We will remove them from cache after installing

File: npm_test.go (hunk 2 above)
Comment (udaykb2) [reply npm_test.go]: 1. For optional + regular dependency missing -> Cache is missing
2. For Bundle + Peer -> integrity value is missing
